ipykernel/kernelbase.py

```diff
@@ -899,6 +894,41 @@ def _input_request(self, prompt, ident, parent, password=False):
             raise EOFError
         return value
 
+    def _wait_input_request_reply(self):
+        """Wait for an input request reply.
+
+        Raises
+        ------
+        KeyboardInterrupt if a keyboard interrupt is recieved.
+        """
+        # Await a response.
+        reply = None
+        while reply is None:
+            try:
+                reply = self._input_request_loop_step()
+            except Exception:
+                self.log.warning("Invalid Message:", exc_info=True)
+            except KeyboardInterrupt:
+                # re-raise KeyboardInterrupt, to truncate traceback
+                raise KeyboardInterrupt
+        return reply
+
+    def _input_request_loop_step(self):
+        """Do one step of the input request loop."""
+        # Allow GUI event loop to update
+        if sys.version_info >= (3, 4):
```

Comment:
I think the minimal supported version in the master branch is 3.5 (please check that in `setup.py`)

Reply:
This doubles as a test for python 2